core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,317 @@
+    await this.select(viewContributorCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewRoleUserResult,
+      element => element.innerText
+    );
+    if (displayedUsers.includes(username)) {
+      throw new Error(
+        `${username} has the right to review question!`);
+    }
+  }
+
+  /**
+   * Function to display reviewing questions rights by role
+   * and check if the user is not displayed as a question submitter
+   * @param {string} usename - the user expected to not be displayed.
+   */
+  async verifyQuestionSubmittersExcludeUser(username) {
+    await this.select(viewContributorFilterMethodSelect, roleMethodValue);
+    await this.select(viewContributorCategorySelect, submitQuestionRightValue);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewRoleUserResult);
+    const displayedUsers = await this.page.$eval(
+      viewRoleUserResult,
+      element => element.innerText
+    );
+    if (displayedUsers.includes(username)) {
+      throw new Error(
+        `${username} has the right to submit question!`);
+    }
+  }
+
+  /**
+   * Function to display contribution rights by user
+   * and check if the user has the right to review questions
+   * @param {string} username - the username of the user to view.
+   */
+  async verifyUserCanReviewQuestions(username) {
+    await this.select(viewContributorFilterMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewContributorReviewQuestionsResult);
+    const displayedUsername = await this.page.$eval(
+      viewContributorReviewQuestionsResult,
+      element => element.innerText);
+    if (displayedUsername.includes('Not-allowed')) {
+      throw new Error(
+        `${username} does not have rights for reviewing questions!`);
+    } else {
+      showMessage(
+        `${username} has rights for reviewing questions.`);
+    }
+  }
+
+  /**
+   * Function to display contribution rights by user
+   * and check if the user has the right to submit questions
+   * @param {string} username - the username of the user to view.
+   */
+  async verifyUserCanSubmitQuestions(username) {
+    await this.select(viewContributorFilterMethodSelect, usernameMethodValue);
+    await this.type(viewContributerUsernameInput, username);
+    await this.clickOn(viewContributorSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+
+    await this.page.waitForSelector(viewContributorSubmitQuestionResult);
+    const displayedUsername = await this.page.$eval(
```

Comment:
This should be plural since you're retrieving a list and calling "includes" on it, right?

Also, something seems off about the naming here. Why would a list of "displayed usernames"  include the text "Not-allowed"? Can you use a variable name that describes what the list actually contains?

Ditto elsewhere.

Reply:
@seanlip, this function checks if the specific user is allowed to submit questions. The website should show a table like the following:
<img width="610" alt="image" src="https://github.com/oppia/oppia/assets/122439444/f6a948ce-2918-4bf6-bfa4-cef66c2941d9">
The variable displayedUsername should store the value of the result (under "Submit Questions"), which should be either "allowed" or "not-allowed", depends on whether the user is allowed to submit questions. I will change the variable name to displayResult as it is confusing.

Reply:
Thanks -- yup the current name is confusing and doesn't describe the div properly. But displayResult is pretty vague. I suggest that you extract the value from the list and call it something like questionReviewStatusForUser or similar.

Reply:
Done